src/Test/L0/Worker/TaskManagerL0.cs

```diff
@@ -131,5 +132,153 @@ private Stream GetZipStream()
                 }
             }
         }
+
+        //Test the cancellation flow: interrupt download task via HostContext cancellation token.
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public async void TestCancellationOfEnsureTasksExist()
+        {
+            //Arrange
+            using (var hc = new TestHostContext(this))
+            {
+                hc.SetSingleton<IJobServer>(_jobServer.Object);
+                hc.SetSingleton<IConfigurationStore>(_configurationStore.Object);
+                var taskManager = new TaskManager();
+                taskManager.Initialize(hc);
+
+                _configurationStore.Setup(x => x.GetSettings()).Returns(
+                 new AgentSettings
+                 {
+                     WorkFolder = WorkFolderName
```

Comment:
FYI - We should be generating a random folder that gets cleaned up each time. Multiple tests run in parallel and the _work folder is common to other classes.

I already changed it for this file in my branch. I'll rebase after this PR goes through.